luaui/Widgets/gui_gridmenu.lua

```diff
@@ -1160,22 +1173,30 @@ local function gridmenuKeyHandler(_, _, args, _, isRepeat)
 
 			local opts
 
-			if ctrl then
+			if quantity < 0 then
+				quantity = quantity * -1
 				opts = { "right" }
 				Spring.PlaySoundFile(CONFIG.sound_queue_rem, 0.75, "ui")
 			else
 				opts = { "left" }
 				Spring.PlaySoundFile(CONFIG.sound_queue_add, 0.75, "ui")
-			end
 
+				--if quantity is more than 100, more than 20 or more than 5 then use engine logic for better performance (fewer for loops inside queueUnit())
+				if alt then
+					quantity = multiQueue(uDefID,quantity,100,{ "alt","left","ctrl","shift" })
+					quantity = multiQueue(uDefID,quantity,20,{ "alt","left","ctrl" })
+					quantity = multiQueue(uDefID,quantity,5,{ "alt","left","shift" })
+				else
+					quantity = multiQueue(uDefID,quantity,100,{ "left","ctrl","shift" })
+					quantity = multiQueue(uDefID,quantity,20,{ "left","ctrl" })
+					quantity = multiQueue(uDefID,quantity,5,{ "left","shift" })
+				end
```

Comment:
the two branches look the same, how about
```suggestion
				quantity = multiQueue(uDefID,quantity,100,{ "ctrl","shift", alt and "alt" })
				quantity = multiQueue(uDefID,quantity,20,{ "ctrl", alt and "alt" })
				quantity = multiQueue(uDefID,quantity,5,{ "shift", alt and "alt" })
```

Reply:
Yea looks cleaner. Done.